genie/load.py

```diff
@@ -308,42 +307,28 @@ def store_database(
 
     Args:
         syn (synapseclient.Synapse): Synapse object
-        database_synid (str): Synapse Id of the Synapse table
-        col_order (List[str]): The ordered column names to be saved
+        database_table_synid (str): Synapse Id of the Synapse table
         all_updates (pd.DataFrame): rows to be appended and/or updated
         to_deleted_rows (pd.DataFrame): rows to be deleted
     """
-    storedatabase = False
-    update_all_file = tempfile.NamedTemporaryFile(
-        dir=process_functions.SCRIPT_DIR, delete=False
-    )
-    with open(update_all_file.name, "w") as updatefile:
-        # Must write out the headers in case there are no appends or updates
-        updatefile.write(",".join(col_order) + "\n")
-        if not all_updates.empty:
-            """
-            This is done because of pandas typing.
-            An integer column with one NA/blank value
-            will be cast as a double.
-            """
-            updatefile.write(
-                all_updates[col_order]
-                .to_csv(index=False, header=None)
-                .replace(".0,", ",")
-                .replace(".0\n", "\n")
+    # get the table entity
+    table_entity = syn.get(database_table_synid)
+    # upsert table with new and updated rows
+    if not all_updates.empty:
+        # TEST: convert to csv to enforce float_format="%.12g" in store_rows
+        with tempfile.TemporaryDirectory() as tmpdir:
+            tmpfile = os.path.join(tmpdir, "all_updates.csv")
+            all_updates.to_csv(tmpfile)
+            Table(id=database_table_synid).store_rows(tmpfile)
```

Comment:
Seeing an error with storing rows in this failed integration tests: https://github.com/Sage-Bionetworks/Genie/actions/runs/18978208245/job/54204110783